m4/ax_check_libsemigroup.m4

```diff
@@ -0,0 +1,52 @@
+dnl handle libsemigroups checks
+dnl
+dnl if --with-included-libsemigroups is supplied, always use the included version
+dnl otherwise, use an external version if it is known to pkg-config and is new enough
+dnl failing the latter, use the included version
+dnl
+AC_DEFUN([AX_CHECK_LIBSEMIGROUPS], [
+  AC_ARG_WITH([included-libsemigroups],
+	      [AC_HELP_STRING([--with-included-libsemigroups],
+			      [use the libsemigroups included here])])
+  REQUI_LIBSEMIGROUPS_VERSION="$(cat .LIBSEMIGROUPS_VERSION)"
+  need_included_libsemigroups=yes
+  PKG_CHECK_MODULES([LIBSEMIGROUPS], [libsemigroups >= $REQUI_LIBSEMIGROUPS_VERSION], 
+  	[need_included_libsemigroups=no],
+  	[need_included_libsemigroups=yes])
+
+  if test "$need_included_libsemigroups" = yes; then
+     AC_MSG_NOTICE([Using included libsemigroups.])
+     if test "$with_included_libsemigroups" = no; then
+	with_included_libsemigroups=yes
+     fi
+  fi
+
+  if test "$with_included_libsemigroups" = yes;  then
+  	AC_CHECK_FILE(
+   		[libsemigroups/src/semigroups.h],
+   		[],
+   		[AC_MSG_ERROR([libsemigroups is required, clone or download the repo from
+    			https://github.com/james-d-mitchell/libsemigroups into this directory])])
+
+	AC_CHECK_FILE(
+   		[libsemigroups/VERSION],
+   		[],
+   		[AC_MSG_ERROR([libsemigroups version 0.6.1 or higher is required])])
```

Comment:
It'd be better if the version number wasn't hard coded here, but I appreciate that this is the same as it was previously (and not a change that you are introducing).

Reply:
So what should it be, instead of 0.6.1?

Reply:
The value in the file `.LIBSEMIGROUPS_VERSION` I think.

Reply:
fixed by bab804e